src/main/java/com/jnape/palatable/lambda/functions/builtin/fn4/Splice.java

```diff
@@ -0,0 +1,52 @@
+package com.jnape.palatable.lambda.functions.builtin.fn4;
+
+import com.jnape.palatable.lambda.functions.Fn1;
+import com.jnape.palatable.lambda.functions.Fn2;
+import com.jnape.palatable.lambda.functions.Fn3;
+import com.jnape.palatable.lambda.functions.Fn4;
+import com.jnape.palatable.lambda.internal.iteration.SplicingIterator;
+
+/**
+ * Return an {@link Iterable} that is the result of splicing all of the elements of an {@link Iterable} into
+ * another {@link Iterable} at a given position, optionally replacing a number of elements in the original {@link Iterable}.
+ * <p>
+ * If <code>startIndex</code> exceeds the size of <code>original</code>, <code>replacement</code> is concatenated to the end.
+ * <p>
+ * Note that <code>splice</code> is equivalent to, but more efficient than
+ * <code>take(startIndex, original) ++ replacement ++ drop(startIndex + replaceCount, original)</code>.
+ *
+ * @param <A> The Iterable element type
+ */
+public final class Splice<A> implements Fn4<Integer, Integer, Iterable<A>, Iterable<A>, Iterable<A>> {
+
+    private static final Splice<?> INSTANCE = new Splice<>();
+
+    private Splice() {
+    }
+
+    @Override
+    public Iterable<A> checkedApply(Integer startIndex, Integer replaceCount, Iterable<A> replacement, Iterable<A> original) {
+        return () -> new SplicingIterator<>(startIndex, replaceCount, replacement.iterator(), original.iterator());
```

Comment:
This is not stack-safe. This isn't a deal breaker, but have you considered creating a `SplicingIterable` that stores an internal `Iterable` of tuples representing all the splicing segments indices and segments, then having the `SplicingIterator` fold through them?

Reply:
[`RateLimitingIterable`](https://github.com/palatable/lambda/blob/master/src/main/java/com/jnape/palatable/lambda/internal/iteration/RateLimitingIterable.java) is an example of what I'm referring to. Again, there are non-stack-safe functions in lambda, no question, but ideally things would be stack-safe where possible, easy, and likely to cause problems otherwise. If the stack-safe implementation seems elusive, I'll merge this in and we can work on it afterwards if we think it's worth it.

Reply:
I had started out with a `SplicingIterable` initially, but then decided stack safety might be overkill.  But now I can think of a couple practical situations where stack safety would be beneficial.  I will take a crack at reimplementing this to be stack safe.

Reply:
Sorry, I meant to follow up with a comment acknowledging your taking "a crack at reimplementing this to be stack safe". Awesome! I'll hold off on the PR merge then until that reimplementation.

Thanks @kschuetz.

Reply:
Any chance of this happening anytime soon @kschuetz? If not, I'd rather just close this PR and wait for you to open a new one later.

Reply:
Not sure how soon it is going to be.  Feel free to go ahead and close, and I'll open a new one later.